### PR TITLE
chore: fix type errors in IPC code

### PR DESCRIPTION
**Pull request by mackrorysd**
## Description

This wasn't caught in the PR, but I was able to reproduce it locally after resetting my Conda environment.

## Test Plan

`make -C harness check` passes now, whereas it doesn't in a clean Conda environment (with only the top-level requirements.txt installed) before.

### CODE REVIEW

**Comment by mackrorysd** [comment]
Hmm, so the linter appears to be unhappy with this in pre-commits, whereas in post-commits it's unhappy with the current state: https://app.circleci.com/pipelines/github/determined-ai/determined/9321/workflows/3bbdca53-1600-4af7-959c-8454e19ac10d/jobs/258493.

**Comment by rb-determined-ai** [comment]
Is it maybe a matter of which version of `zmq` is installed?

**Comment by mackrorysd** [comment]
> Is it maybe a matter of which version of `zmq` is installed?

Could be. I couldn't reproduce this locally until I reset my conda environment (which had already been reset in the last week) and pip installed the top-level requirements.txt. Then I reproduced it. If I go back to before my commit and do the same, I'm on pyzmq 22.0.0, which is the same version I had while reproducing it. I wonder if there's some cache involved for pre-commits that doesn't get used later?

**Comment by dzhu** [comment]
I'm beginning to suspect this is another new-resolver-in-pip-20.3 issue.

**Comment by dzhu** [comment]
Oh, this probably all just happened because pyzmq _just_ released 22.0.0. https://pypi.org/project/pyzmq/#history :zany_face:

File: harness/setup.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "packaging",
         "numpy>=1.16.2",
         "psutil",
-        "pyzmq>=18.1.0",
+        "pyzmq>=22.0.0",
```
Comment (rb-determined-ai) [inline harness/setup.py:26]: We don't need our users to pin this so strictly?  Can we put the requirement somewhere else, that will affect CI but not users?

Comment (dzhu) [reply harness/setup.py:26]: Ah, makes sense. Sean had it there and I suggested the move to `setup.py`, sorry.

**Review by rb-determined-ai — APPROVED** [review]

**Pull request merged**